sql/stats: TestQuantileOpsRandom failed

sql/stats.TestQuantileOpsRandom [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8659827?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8659827?buildTab=artifacts#/) on master @ [9ea0ef2f6a8e31e22c7350de5cacd9c5992d07a0](https://github.com/cockroachdb/cockroach/commits/9ea0ef2f6a8e31e22c7350de5cacd9c5992d07a0):


```
=== RUN   TestQuantileOpsRandom
--- FAIL: TestQuantileOpsRandom (0.00s)
=== RUN   TestQuantileOpsRandom/2
    quantile_test.go:1415: seed 8933964152301902779 quantile [{0 -6721.87283596984} {0.003542103310572345 -11149.825264181563} {0.0317672629697891 8024.953568287907} {0.07690317062589916 3831.6076571769554} {0.1269800259929379 -6603.896844467807} {0.1282500781980292 1703.4417721462803} {0.15859318641291037 6127.068876077339} {0.21630555409568702 -11149.814600794209} {0.21630555409568702 -6721.87283596984} {0.25493764378214445 8024.953568287907} {0.29231554192810716 -6603.896844467807} {0.3381104558302833 3831.6076571769554} {0.36926402967725525 -6603.896844467807} {0.36926402967725525 3831.6076571769554} {0.4223648974701217 -6603.896844467807} {0.47805113025608226 -11149.825264181563} {0.5069130009480403 -11149.814600794209} {0.5155091579890764 -11149.814600794209} {0.7326101578209553 -11149.825264181563} {0.7936625966248472 8024.953568287907} {0.8093791217157414 -11149.825264181563} {0.8350623037790734 -11149.814600794209} {0.8443649791020295 1703.4417721462803} {0.8585879759919895 -11149.825264181563} {0.8806311213632848 -11149.825264181563} {0.8806311213632848 -6721.87283596984} {0.8810526992981722 1703.4417721462803} {0.9107175648846253 -6603.896844467807} {0.9164057228287454 -6721.87283596984} {0.9164057228287454 -11149.825264181563} {0.9222262267178467 1703.4417721462803} {1 -6603.896844467807}] incorrect intSqFixed 5.576890354212272e+07 expected 5.576890354079543e+07
    --- FAIL: TestQuantileOpsRandom/2 (0.00s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestQuantileOpsRandom.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24422

--- a/pkg/sql/stats/quantile_test.go
+++ b/pkg/sql/stats/quantile_test.go
@@ -1366,7 +1366,7 @@ func TestQuantileOps(t *testing.T) {
 
 // TestQuantileOpsRandom tests basic operations on random quantile functions.
 func TestQuantileOpsRandom(t *testing.T) {
-	const delta = 1e-3
+	const delta = 1e-2
 	rng, seed := randutil.NewTestRand()
 	for i := 0; i < 5; i++ {
 		t.Run(strconv.Itoa(i), func(t *testing.T) {

